src/bot/lib/query.js

```diff
@@ -11,56 +9,57 @@ export function abort(...args) {
   return error;
 }
 
-// Find matching skills for the given search term.
-export function findSkillByName(token, search) {
-  return query.skillByName({token, search}).then(matches => {
-    let exact;
-    if (matches.length > 0) {
-      exact = matches.find(m => m.name.toLowerCase() === search.toLowerCase());
-    }
-    return {
-      // All matches.
-      matches,
-      // The "best" match. Might not be exact.
-      match: exact || matches[0],
-      // An exact match. (case-insensitive)
-      exact: exact || null,
-    };
-  });
+// Find matching items for the given search term.
+export function parseMatches(matches, search) {
+  let exact;
+  if (matches.length > 0) {
+    exact = matches.find(m => m.name.toLowerCase() === search.toLowerCase());
+  }
+  return {
+    search,
+    // All matches.
+    matches,
+    // The "best" match. Might not be exact.
+    match: exact || matches[0],
+    // An exact match. (case-insensitive)
+    exact: exact || null,
+  };
 }
 
-// Find the best match for the given search term, and complain if necessary.
-export function findSkillAndHandleErrors(token, search) {
+// processes matches and returns output for dialogue
+export function returnMatches(matchResults) {
+  const {search, matches, match, exact} = matchResults;
   const output = [];
-  return findSkillByName(token, search).then(({matches, match, exact}) => {
-    if (matches.length === 0) {
-      throw abort(`_No matches found for skill "${search}"._`);
-    }
-    else if (matches.length === 1) {
-      output.push(`_You specified "${search}", which matches: *${matches[0].name}*._`);
+
+  if (matches.length === 0) {
+    throw abort(`_No matches found for "${search}"._`);
+  }
+  else if (matches.length === 1) {
+    output.push(`_You specified "${search}", which matches: *${matches[0].name}*._`);
+  }
+  else {
+    const list = matches.map(item => item.name).join(', ');
+    output.push(`_Multiple matches were found: ${list}._`);
+    if (exact) {
+      output.push(`_You specified "${search}", which matches: *${exact.name}*._`);
     }
     else {
-      const skillsList = matches.map(skill => skill.name).join(', ');
-      output.push(`_Multiple matches were found: ${skillsList}._`);
-      if (exact) {
-        output.push(`_You specified "${search}", which matches: *${exact.name}*._`);
-      }
-      else {
-        throw abort(`_You specified "${search}", which is ambiguous. Please be more specific._`);
-      }
+      throw abort(`_You specified "${search}", which is ambiguous. Please be more specific._`);
```

Comment:
Throwing here will prevent the messages in `output` from appearing alongside the error. That is why this method was originally created. If we populate the errors in an errors array, we can check outside this function if there was an error and act accordingly (`throwIfMatchErrors` combines errors and output).